py/desispec/desi_proc_dashboard.py

```diff
@@ -0,0 +1,383 @@
+            nights=listdir(os.getenv('DESI_SPECTRO_REDUX')+'/daily/exposures/')
+            nights=[int(x) for x in nights]
+        else:
+            try:
+                loc=locals()
+                cmd='nights='+args.nights
+                a=exec(cmd)
+                nights=loc['nights']
+            except:
+                nights=[]
+
+        tonight=self.what_night_is_it()
+        if not tonight in nights:
+            nights.append(tonight)
+        nights.sort(reverse=True)
+        if int(args.n_night)<=len(nights):
+            nights=nights[0:int(args.n_night)-1]
+
+        prod_dir=args.prod_dir # base directory of product
+        self.output_dir=args.output_dir # Portal directory for output html files
+        self.output_url=args.output_url # corresponding URL
+
+        ######################################################################
+        ## sub directories. Should not change if generated by the same code ##
+        ## that follows the same directory strucure ##
+        ######################################################################
+        self.data_dir=prod_dir+"/spectro/data/"
+        self.log_dir=prod_dir+"/spectro/redux/daily/run/logs"
+        self.redux_dir=prod_dir+"/spectro/redux/daily"
+
+        strTable=self._initialize_page()
+
+        for night in nights:
+            #print(night)
+            stat_night=self.calculate_one_night(night)
+            ####################################
+            #### Table for individual night ####
+            ####################################
+            strTable=strTable+self._add_html_table(stat_night,str(night))
+            
+        timestamp=time.strftime("%Y-%m-%d %H:%M:%S",time.localtime())
+        #print(timestamp)
+        running=self.check_running()
+        strTable=strTable+"<div style='color:#00FF00'>"+timestamp+" "+"desi_dailyproc running: "+running+"</div>"
+        strTable=strTable+self._add_js_script1()
+        strTable=strTable+"</html>"
+        hs=open(self.output_dir+"desi_proc_dashboard.html",'w')
+        hs.write(strTable)
+        hs.close()
+
+        ##########################
+        #### Fix Permission ######
+        ##########################
+        cmd="chmod -R a+xr "+self.output_dir
+        os.system(cmd)
+
+    def _init_parser(self,parser):
+        parser.add_argument('-n','--nights', type=str, default = None, required = False, help="nights to monitor")
+        parser.add_argument('-nn','--n_night', type=str, default = None, required = False, help="all:all nights. ifdigit: the last n nights.")
+        parser.add_argument('-pd','--prod_dir', type=str, default = None, required = True, help="product base directory")
+        parser.add_argument('-od','--output_dir', type=str, default = None, required = True, help="output portal directory for the html pages ")
+        parser.add_argument('-ou','--output_url', type=str, default = None, required = True, help="output portal directory url ")
```

Comment:
It is non-standard python options to have a single-dash options that requires 2 letters, e.g. `-nn`.  It is ok to have options that only have double-dash `--output_url` without having any single-dash option.

Let's make default `args.prod_dir = desispec.io.specprod_root()` and add a new option `args.rawdata_dir` with default `desispec.io.rawdata_root()`.

Reply:
prod_dir is in fact useless because the code only checks the redux directory, sorry about the confusion...

Reply:
I remove all '--xx' options.

Reply:
Realize that the code does not use raw data, thus prod_dir is changed to point to desispec.io.specrpod_root() instead of the raw data directory.